### PR TITLE
Configuring Pod Security for Workloads

### PR DESCRIPTION
**Pull request by rashedkvm**
# Which other branches do you want a technical writer to cherry-pick this PR to (if any)?

`main` Branch only
 
It's best to PR to the most recent relevant branch and leave all the cherry-picking to the
writer, except where earlier branches require factual changes to the content.
For more information about the branches, see https://github.com/pivotal/docs-tap#branches

### CODE REVIEW

File: scc/install-ootb-templates.hbs.md [hunk 1]
```diff
@@ -45,6 +82,14 @@ To install Out of the Box Templates:
 
     ```yaml
     excluded_templates: []
+    podintent_secuirty_context:
```
Comment (rszumlakowski) [inline scc/install-ootb-templates.hbs.md:85]:
```suggestion
    podintent_security_context:
```

File: security-and-compliance/pod-security-for-workloads.hbs.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+"security-context" which overrides the default set by the platform operator. 
+
+## <a id="platform-default"></a> Setting a default security context
+
+A default workload security context can be optionally configured using tap-values.yaml.
+The YAML file lists the security context configurable fields. 
+
+```yaml
+ootb_templates:
+  podintent_secuirty_context:
```
Comment (rszumlakowski) [inline security-and-compliance/pod-security-for-workloads.hbs.md:26]:
```suggestion
  podintent_security_context:
```

File: scc/install-ootb-templates.hbs.md [hunk 3]
```diff
@@ -25,17 +25,54 @@ To install Out of the Box Templates:
...
+                                                                                                                                                                RuntimeDefault - the container runtime default profile should be used.
+                                                                                                                                                                Unconfined - no profile should be applied.
+    podintent_security_context.privileged                                                                                                               boolean  Run container in privileged mode. Defaults to false.
+    podintent_security_context.procMount                                                                                                                string   procMount denotes the type of proc mount to use for the containers.
+    ca_cert_data                                                                                                                                        string   PEM encoded certificate data for the image registry where the image will be
```
Comment (rszumlakowski) [inline scc/install-ootb-templates.hbs.md:70]:
```suggestion
    ca_cert_data                                                                                                                                        string   PEM-encoded certificate data for the image registry where the image will be
```

File: scc/install-ootb-templates.hbs.md [hunk 4]
```diff
@@ -25,17 +25,54 @@ To install Out of the Box Templates:
...
+                                                                                                                                                                pushed to.
+    excluded_templates                                                                                                                                  array    List of templates to exclude from the installation (e.g. ['git-writer'])
+    iaas_auth                                                         false                                                                             boolean  Enable the use of IAAS based authentication for imgpkg
```
Comment (rszumlakowski) [inline scc/install-ootb-templates.hbs.md:73]:
```suggestion
    iaas_auth                                                         false                                                                             boolean  Activate the use of IAAS based authentication for imgpkg
```

File: scc/install-ootb-templates.hbs.md [hunk 5]
```diff
@@ -25,17 +25,54 @@ To install Out of the Box Templates:
...
+    kubernetes_distribution                                                                                                                             string   Optional: Type of K8s infrastructure being used. Supported Values: 'openshift',
```
Comment (rszumlakowski) [inline scc/install-ootb-templates.hbs.md:74]:
```suggestion
    kubernetes_distribution                                                                                                                             string   Optional: Type of K8S infrastructure being used. Supported Values: 'openshift',
```

File: security-and-compliance/pod-security-for-workloads.hbs.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+This topic describes the various configuration options that are available for applying security 
+contexts to pods that are responsible for running supply chain workloads. To apply security contexts
+to pods responsible for running supply chain workloads, organizations can leverage various
+configuration options. With this approach, security policies can be configured and specific 
+security requirements of applications running on a Kubernetes cluster can be met.
+
+## <a id="config-options"></a> Configuration options
+
+There are two methods to provide pod security context for running supply chain workloads.
```
Comment (rszumlakowski) [inline security-and-compliance/pod-security-for-workloads.hbs.md:11]:
```suggestion
There are two methods to provide the pod security context for running supply chain workloads. 
```

File: security-and-compliance/pod-security-for-workloads.hbs.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+- Platform operators can set a default security context for all supply chain workloads via 
+TAP values YAML file during TAP installation or upgrade. 
+
+- App operators can set workload security context via the workload parameter name 
+"security-context" which overrides the default set by the platform operator. 
+
+## <a id="platform-default"></a> Setting a default security context
+
+A default workload security context can be optionally configured using tap-values.yaml.
```
Comment (rszumlakowski) [inline security-and-compliance/pod-security-for-workloads.hbs.md:21]:
```suggestion
A default workload security context can be optionally configured using the "tap-values.yaml" file during TAP installation or upgrade.
```

File: security-and-compliance/pod-security-for-workloads.hbs.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+      drop: # string array - Removed capabilities
+    readOnlyRootFilesystem: # boolean - Whether this container has a read-only root filesystem.
+    seLinuxOptions:
+      level: # string - Level is SELinux level label that applies to the container.
+      role: # string - Role is a SELinux role label that applies to the container.
+      type: # string - Type is a SELinux type label that applies to the container.
+      user: # string - User is a SELinux user label that applies to the container.
+    seccompProfile:
+      localhostProfile: # string - LocalhostProfile indicates a profile defined in a file on the node should be used. The profile must be preconfigured on the node to work. Must be a descending path, relative to the kubelet's configured seccomp profile location. Must only be set if type is "Localhost".
+      type: # string - Type indicates which kind of seccomp profile will be applied. Valid options are, Localhost - a profile defined in a file on the node should be used. RuntimeDefault - the container runtime default profile should be used. Unconfined - no profile should be applied.
```
Comment (rszumlakowski) [inline security-and-compliance/pod-security-for-workloads.hbs.md:46]:
```suggestion
      type: # string - Type indicates which kind of seccomp profile will be applied. Valid options are: "Localhost" - a profile defined in a file on the node should be used, "RuntimeDefault" - the container runtime default profile should be used, "Unconfined" - no profile should be applied.
```

File: security-and-compliance/pod-security-for-workloads.hbs.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    procMount: # string - procMount denotes the type of proc mount to use for the containers.
+```
+
+To view the possible configuration setting for the `ootb-templates` package,
+please see [Installing out of the box templates](../scc/install-ootb-templates.hbs.md)
+
+## <a id="workload-config"></a> Setting a Workload security context
+
+Suppply chain workload security context can be optionally configured using the workload parameter 
+name "security-context". Security context configured via workload *overrides* deafult security
```
Comment (rszumlakowski) [inline security-and-compliance/pod-security-for-workloads.hbs.md:57]:
```suggestion
name "security-context". Security context configured via workload *overrides* default security 
```

File: security-and-compliance/pod-security-for-workloads.hbs.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+metadata:
+  name: tanzu-java-web-app
+  labels:
+    apps.tanzu.vmware.com/workload-type: web
+    app.kubernetes.io/part-of: tanzu-java-web-app
+spec:
+  params:
+  - name: annotations
+    value:
+      autoscaling.knative.dev/minScale: "1"
```
Comment (rszumlakowski) [inline security-and-compliance/pod-security-for-workloads.hbs.md:74]: Is this parameter applicable to this example?

File: security-and-compliance/pod-security-for-workloads.hbs.md [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+        - ALL
+  source:
+    git:
+      url: https://github.com/vmware-tanzu/application-accelerator-samples
+      ref:
+        tag: tap-1.7.0
+    subpath: tanzu-java-web-app
+```
+
+Alternatively, suppply chain workload security context can be set when creating workload using the
```
Comment (rszumlakowski) [inline security-and-compliance/pod-security-for-workloads.hbs.md:93]:
```suggestion
Alternatively, the suppply chain workload security context can be set when creating workload using the 
```

File: workloads/troubleshoot-workload.hbs.md [hunk 12]
```diff
@@ -0,0 +1,24 @@
+# Troubleshoot Workload
+
+This topic gives you guidance about how to troubleshoot issues with Supply Chain Workload.
+
+## <a id="pod-security-workload"></a> Errors when the security context is misconfigured
+
+If workload in the supply chain fails to progess with error status shown below, it is prossibly
```
Comment (rszumlakowski) [inline workloads/troubleshoot-workload.hbs.md:7]:
```suggestion
If workload in the supply chain fails to progess with the error status shown below, it is possibly 
```

File: workloads/troubleshoot-workload.hbs.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
+# Troubleshoot Workload
+
+This topic gives you guidance about how to troubleshoot issues with Supply Chain Workload.
+
+## <a id="pod-security-workload"></a> Errors when the security context is misconfigured
+
+If workload in the supply chain fails to progess with error status shown below, it is prossibly 
+the workload  security context is misconfigured. For example:
```
Comment (rszumlakowski) [inline workloads/troubleshoot-workload.hbs.md:8]:
```suggestion
If workload in the supply chain fails to progess with the error status shown below then it is possible that
the workload security context is misconfigured. For example:
```

File: workloads/troubleshoot-workload.hbs.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+     must set securityContext.allowPrivilegeEscalation=false)'
+   reason: HealthyConditionRule
+   status: "False"
+   type: ResourcesHealthy
+```
+
+This error means the supply chain created workload deployment does not meet 
+[pod security standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/)  
+used in the cluster. Please see [Configuring Pod Security for Workloads](/security-and-compliance/pod-security-for-workloads.hbs.md)
+for setting supply chain workload security context.
```
Comment (rszumlakowski) [inline workloads/troubleshoot-workload.hbs.md:24]:
```suggestion
This error means that the workload Deployment created by supply chain does not meet 
[pod security standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/)  
used in the cluster. Please see [Configuring Pod Security for Workloads](/security-and-compliance/pod-security-for-workloads.hbs.md)
for setting the supply chain workload security context.
```

File: security-and-compliance/pod-security-for-workloads.hbs.md (hunk 10 above)
Comment (rashedkvm) [reply security-and-compliance/pod-security-for-workloads.hbs.md:74]: It does not impact the security context of the deployment

**Review by rszumlakowski — APPROVED** [review]

**Pull request merged**